tensorflow/core/graph/mkl_layout_pass_test.cc

```diff
@@ -671,65 +812,84 @@ TEST_F(MklLayoutPassTest, NodeMerge_PadWithConv2D_Common_Input) {
-      " attr { key: 'use_cudnn_on_gpu' value { b: false } }"
-      " attr { key: 'strides'          value { list: {i: 1, i:1, i:1, i:1} } }"
-      " attr { key: 'padding'          value { s: 'VALID' } }"
-      " attr { key: 'dilations'        value { list: {i: 1, i:1, i:1, i:1} } }"
-      " input: ['C', 'A'] }"
-      "node { name: 'Z' op: 'Output2'"
-      " input: ['C', 'E']}");
-  EXPECT_EQ(DoMklLayoutOptimizationPass(),
-            "A(Input);B(Int32Input);C(Pad);DMT/_0(Const);DMT/_1(Const);"
-            "E(_MklConv2D);Z(Output2)|A->C;A->E:1;B->C:1;C->E;C->Z;"
-            "C:control->DMT/_0:control;C:control->DMT/_1:control;"
-            "DMT/_0->E:2;DMT/_1->E:3;E->Z:1");
-}
+#define REGISTER_TEST(NAME, T, INPUT, OUTPUT)                              \
+  TEST_F(MklLayoutPassTest, NAME##_##T) {                                  \
+    DCHECK_EQ(kTensorOrdering, MklTfTensorOrdering::TENSORS_CONTIGUOUS);   \
+    InitGraph("node { name: 'A' op: '" #INPUT "'}"                         \
+              "node { name: 'B' op: 'Int32Input'}"                         \
+              "node { name: 'C' op: 'Pad'"                                 \
+              " attr { key: 'T'                value { type: " #T " } }"   \
+              " attr { key: 'Tpaddings'        value { type: DT_INT32 } }" \
+              " input: ['A', 'B']}"                                        \
+              "node { name: 'E' op: 'Conv2D'"                              \
+              " attr { key: 'T'                value { type: " #T " } }"   \
+              " attr { key: 'data_format'      value { s: 'NHWC' } }"      \
+              " attr { key: 'use_cudnn_on_gpu' value { b: false } }"       \
+              " attr { key: 'strides'          value { list: {i: 1, i:1, " \
+              "i:1, i:1} } }"                                              \
+              " attr { key: 'padding'          value { s: 'VALID' } }"     \
+              " attr { key: 'dilations'        value { list: {i: 1, i:1, " \
+              "i:1, i:1} } }"                                              \
+              " input: ['C', 'A'] }"                                       \
+              "node { name: 'Z' op: '" #OUTPUT "'"                         \
+              " input: ['C', 'E']}");                                      \
+    EXPECT_EQ(DoMklLayoutOptimizationPass(),                               \
+              "A(" #INPUT                                                  \
+              ");B(Int32Input);C(Pad);DMT/_0(Const);DMT/_1(Const);"        \
+              "E(_MklConv2D);Z(" #OUTPUT                                   \
+              ")|A->C;A->E:1;B->C:1;C->E;C->Z;"                            \
+              "C:control->DMT/_0:control;C:control->DMT/_1:control;"       \
+              "DMT/_0->E:2;DMT/_1->E:3;E->Z:1");                           \
+  }
+REGISTER_TEST(NodeMerge_PadWithConv2D_Common_InOutput, DT_FLOAT, Float32Input,
+              Float32Output2)
```

Comment:
Nit: Would you mind adding a semicolon at the end? Syntactically, this doesn't need it, but it makes it looks cleaner (and consistent with other calls to REGISTER_TEST in this file).

Same for `REGISTER_TEST` in line 848.

Reply:
Done